engines/sci/sound/music.cpp

```diff
@@ -467,6 +467,9 @@ void SciMusic::soundInitSnd(MusicEntry *pSnd) {
 			pSnd->overridePriority = false;
 
 			pSnd->pMidiParser->loadMusic(track, pSnd, channelFilterMask, _soundVersion);
+
+			// When restoring, continue from the correct place
+			pSnd->pMidiParser->jumpToTick(pSnd->ticker);
```

Comment:
Why? We already do this when loading. Also, all the music events need to run when restoring (like instrument changes), otherwise music will sound distorted after loading

Reply:
Why?
Currently, when we save a game while a sound track is playing, and later restore it, the track is starting from the beginning. It's easy to see that in the scenerio of https://bugs.scummvm.org/ticket/10812 - KQ6 "wallflower dance music" - if you save the game few seconds before the music stops, and then restore, it starts from the beginning.

You're correct that sci\engine\savegame.cpp `MusicEntry::saveLoadWithSerializer` has `s.syncAsSint16LE(ticker)` , but it's not enough, as in sci\sound\music.cpp `MusicEntry::onTimer()` there is `ticker = (uint16)pMidiParser->getTick();` which overrides it, to zero, and thus the track starts from the beginning.

Therefore, calling `jumpToTick(pSnd->ticker)`  fixes it.

However, I changed it to be `pSnd->pMidiParser->jumpToTick(pSnd->ticker, true, true, true);` , which (if I understand correctly) should acheive exactly that - fast forward to the correct place, executing all midi controllers in the way, without playing music.
I copied this line, and its comment, from music.cpp, from near the end of `SciMusic::soundPlay`  (line 609 in my version)

Reply:
Hi.
@bluegr , any news?